staging/src/k8s.io/apiserver/pkg/storage/etcd3/lease_manager.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcd3
+
+import (
+	"context"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+)
+
+// leaseManager is used to manage leases requested from etcd. If a new write
+// needs a lease that has similar expiration time to a previous one, the old
+// lease will be reused to reduce the overhead of etcd, since lease operations
+// are expensive.
+type leaseManager struct {
+	// etcd client used to grant leases
+	client *clientv3.Client
+	// the number of buffered leases, i.e., size of the following mappings
+	size int
+	// the mutex used for the following maps
+	leaseMu sync.Mutex
```

Comment:
For consistency with other parts of k8s codebased, can you please call it "lock" ?

Reply:
I followed here in the same package:

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/etcd3/compact.go#L34

If it's better to change it to lock, I am okay to do so.